### ISSUE TITLE
Issue when using JsonTypeInfo/JsonSubTypes

### ISSUE THREAD

**Issue by antechrestos** [immutables/immutables#360]
Hi,

first of all I would like to thank you for your work that helped me a lot. 

However, I met what may be an issue.

[In the code that I made as an example](https://github.com/antechrestos/example-immutable), I have a response class that has a polymorphic payload.

```java
@JsonDeserialize
@Value.Immutable
abstract class _AbstractResponse {

    @JsonProperty("type")
    public abstract String getType();

    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXTERNAL_PROPERTY, property = "type")
    @JsonSubTypes({
            @JsonSubTypes.Type(name = "one", value = PayloadOne.class),
            @JsonSubTypes.Type(name = "two", value = PayloadTwo.class)
    })
    @JsonProperty("payload")
    public abstract Payload getPayload();
}
```

`Payload` is an interface that has two abstract classes that implement it, `_PayloadOne` and `_PayloadTwo`. Each of these implementations are `abstract` and candidate to be immutable:

```java
@JsonDeserialize
@Value.Immutable
abstract class _PayloadOne implements Payload {
    @JsonProperty("propertyOne")
    abstract String getPropertyOne();
    @JsonProperty("propertyTwo")
    abstract String getPropertyTwo();
}
```

```java
@JsonDeserialize
@Value.Immutable
abstract class _PayloadTwo implements Payload {
    @JsonProperty("propertyOne")
    abstract Integer getPropertyOne();
    @JsonProperty("propertyTwo")
    abstract Long getPropertyTwo();
```

Doing the compilation phase generates a code for `AbstractResponse` that fails:

```java
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXTERNAL_PROPERTY, property = "type")
  @JsonSubTypes({@JsonSubTypes.Type(name = "one", value = "<error>"), @JsonSubTypes.Type(name = "two", value = "<error>")})
  @JsonProperty("payload")
  @Override
  public Payload getPayload() {
    return payload;
  }
```
As you can see the `value`atttribute of `JsonSubTypes.Type`annotation indicates an `error`.

Adding the annotations JsonTypeInfo/JsonSubTypes after a first class generation makes it work but it is not a good solution ... 

Thanks!

**Comment by elucash** [comment]
Thank you for reporting this! It was a known issue with Javac and not-yet-generated-types as class literals in nested annotations being dispatched as string `"<error>"`. I've tried to apply a workaround with Javac-specific code, which seems to work. You try get the latest snapshots 2.2.7-SNAPSHOT from oss sonatype repo or build it from sources. Will be released in the next version, in a week or so.

**Comment by antechrestos** [comment]
@elucash : I just made the test rebuilding from your source and it works like a charm.
Looking for the next verion then :smirk: 
Do you want me to close the issue or will it be closed automatically when next version is released?

**Comment by elucash** [comment]
let it hang for while, I usually close issues after the release when compiling a changelog. Thank you for verifying the fix!

**Comment by elucash** [comment]
released in 2.2.7 thank you!

**Comment by antechrestos** [comment]
thanks!
